src/index.ts

```diff
@@ -6,22 +6,21 @@ import type {
 
 // Helper method to wait for a middleware to execute before continuing
 // And to throw an error when an error happens in a middleware
-function initMiddleware(middleware) {
+function initMiddleware(middleware: any) {
```

Comment:
to take advantage of the power of typescript, you'd better switch to the following.
```suggestion
function initMiddleware(middleware: typeof cors) {
```